include/zephyr/kernel.h

```diff
@@ -3625,6 +3625,18 @@ void k_work_queue_start(struct k_work_q *queue,
 			k_thread_stack_t *stack, size_t stack_size,
 			int prio, const struct k_work_queue_config *cfg);
 
+/** @brief Run work queue using calling thread
+ *
+ * This will run the work queue forever, the function never returns
```

Comment:
I noticed that this signature was missing the no-return attribute and then looked at it a bit more. It will return when stopped.

```suggestion
 * This will run the work queue forever unless stopped by @ref k_work_queue_stop.
```

The other thing that could be nice to have here is a `k_timeout_t timeout` but that's probably a separate discussion.

Reply:
Thanks, yeah, forgot to update the docs after properly implementing support for the queue being stopped. Rebased the branch and updated the docs.

Regarding timeout: I could perhaps see the utility in that, but I'm not 100% convinced of the use case. I'd prefer not to mix that in with this PR.